presto-native-execution/presto_cpp/main/functions/remote/PrestoToVeloxRemoteFunctionExpr.cpp

```diff
@@ -0,0 +1,195 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "presto_cpp/main/functions/remote/PrestoToVeloxRemoteFunctionExpr.h"
+#include "presto_cpp/main/common/Configs.h"
+#include "presto_cpp/main/functions/remote/client/RestRemoteClient.h"
+#include "presto_cpp/main/functions/remote/client/VeloxRemoteFunction.h"
+
+using facebook::velox::functions::remote::PageFormat;
+
+namespace facebook::presto::functions::remote::rest {
+namespace {
+// Returns the serialization/deserialization format used by the remote function
+// server. The format is determined by the system configuration value
+// "remoteFunctionServerSerde". Supported formats:
+//   - "presto_page": Uses Presto page format.
+//   - "spark_unsafe_row": Uses Spark unsafe row format.
+// @return PageFormat enum value corresponding to the configured serde format.
+// @throws VeloxException if the configured format is unknown.
+PageFormat getSerde() {
+  static const auto serdeFormat =
+      SystemConfig::instance()->remoteFunctionServerSerde();
+  if (serdeFormat == "presto_page") {
+    return PageFormat::PRESTO_PAGE;
+  } else if (serdeFormat == "spark_unsafe_row") {
+    return PageFormat::SPARK_UNSAFE_ROW;
+  } else {
+    VELOX_FAIL(
+        "Unknown serde name for remote function server: '{}'", serdeFormat);
+  }
+}
+
+// Extracts the schema name from a fully qualified Presto function identifier.
+// The function identifier is expected in the format
+// "namespace.schema.function;TYPE;TYPE". This function returns the substring
+// between the first and second dots in the function name.If the schema cannot
+// be determined, "default" is returned.
+// @param functionId The fully qualified Presto function identifier.
+// @return The extracted schema name or "default".
+std::string getSchemaName(const protocol::SqlFunctionId& functionId) {
```

Comment:
Consider reusing function `getFunctionNameParts` from `FunctionMetadata.cpp`. `getFunctionNameParts` can be moved to `presto-native-execution/presto_cpp/main/common/Utils.cpp`.